packages/opentelemetry-exporter-collector/src/CollectorExporter.ts

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ExportResult } from '@opentelemetry/base';
+import { NoopLogger } from '@opentelemetry/core';
+import { ReadableSpan, SpanExporter } from '@opentelemetry/tracing';
+import { Attributes, Logger } from '@opentelemetry/types';
+
+import * as collectorTypes from './types';
+import { toCollectorSpan } from './transform';
+
+import { onInit, onShutdown, sendSpans } from './platform/index';
+
+/**
+ * Collector Exporter Config
+ */
+export interface CollectorExporterConfig {
+  hostName?: string;
+  logger?: Logger;
+  serviceName?: string;
+  spanAttributes?: Attributes;
+  url?: string;
+}
+
+const defaultServiceName = 'collector-exporter';
```

Comment:
Nit: should this be DEFAULT_SERVICE_NAME, and similar for DEFAULT_COLLECTOR_URL?